### PR TITLE
Revert "Revert "refactor: use cesdk""

### PR DESCRIPTION
**Pull request by grant**
Reverts GoogleCloudPlatform/functions-framework-nodejs#399 (roll forward).

See PR description for details: https://github.com/GoogleCloudPlatform/functions-framework-nodejs/pull/391#issue-1057979568

WANT_LGTM=@matthewrobertson

### CODE REVIEW

File: src/cloud_events.ts [hunk 1]
```diff
@@ -13,7 +13,7 @@
 // limitations under the License.
 
 import * as express from 'express';
-import {CloudEventsContext} from './functions';
+import {CloudEvent} from 'cloudevents';
```
Comment (matthewrobertson) [inline src/cloud_events.ts:16]: we should expose the CESDK interfaces in our APIs not the concrete class.

```
import {CloudEventV1 as CloudEvent} from 'cloudevents';
```

File: src/functions.ts [hunk 2]
```diff
@@ -17,12 +17,7 @@
 
 /* eslint-disable @typescript-eslint/no-explicit-any */
 import * as express from 'express';
-import {CloudEventsContext} from './cloud_event_types/cloud_events_context';
-
-/**
- * @public
- */
-export {CloudEventsContext};
+import {CloudEvent} from 'cloudevents';
```
Comment (matthewrobertson) [inline src/functions.ts:20]: this should be the interface: 

```typescript
import {CloudEventV1 as CloudEvent} from 'cloudevents';
```

File: src/functions.ts [hunk 3]
```diff
@@ -17,12 +17,7 @@
 
 /* eslint-disable @typescript-eslint/no-explicit-any */
 import * as express from 'express';
-import {CloudEventsContext} from './cloud_event_types/cloud_events_context';
-
-/**
- * @public
- */
-export {CloudEventsContext};
```
Comment (matthewrobertson) [inline src/functions.ts:25]: this line was necessary to let developers import the `CloudEvent` interface from the functions framework. You renamed it `CloudEvent` so:

```typescript
export {CloudEvent};
```

Side note: I am surprised api-extractor did catch this.

File: test/function_wrappers.ts [hunk 4]
```diff
@@ -1,7 +1,8 @@
 import * as assert from 'assert';
 import * as sinon from 'sinon';
 import {Request, Response} from 'express';
-import {CloudEventsContext, Context} from '../src/functions';
+import {Context} from '../src/functions';
+import {CloudEvent} from 'cloudevents';
```
Comment (matthewrobertson) [inline test/function_wrappers.ts:5]: probably simpler to reuse the re-exported version of the interface:

```
import {Context, CloudEvent} from '../src/functions';
```

File: test/integration/cloud_event.ts [hunk 5]
```diff
@@ -17,6 +17,7 @@ import * as functions from '../../src/index';
 import * as sinon from 'sinon';
 import {getTestServer} from '../../src/testing';
 import * as supertest from 'supertest';
+import {CloudEvent} from 'cloudevents';
```
Comment (matthewrobertson) [inline test/integration/cloud_event.ts:20]: you shouldn't need this

File: test/integration/cloud_event.ts [hunk 6]
```diff
@@ -39,7 +40,7 @@ const TEST_EXTENSIONS = {
 describe('CloudEvent Function', () => {
   let clock: sinon.SinonFakeTimers;
 
-  let receivedCloudEvent: functions.CloudEventsContext | null;
+  let receivedCloudEvent: CloudEvent | null;
```
Comment (matthewrobertson) [inline test/integration/cloud_event.ts:43]: the `CloudEvent` interface should be exported from the functions framework so:

```typescript
let receivedCloudEvent: functions.CloudEvent | null;
```

File: src/function_wrappers.ts [hunk 7]
```diff
@@ -25,9 +25,9 @@ import {
   Context,
   CloudEventFunction,
   CloudEventFunctionWithCallback,
-  CloudEventsContext,
   HandlerFunction,
 } from './functions';
+import {CloudEvent} from 'cloudevents';
```
Comment (matthewrobertson) [inline src/function_wrappers.ts:30]: you should use the interface not the concrete class implementation. Also you should probably just pull this from `./functions` where it is re-exported:

```typescript
  // ...
  CloudEvent,
} from './functions';
```

File: src/cloud_events.ts (hunk 1 above)
Comment (matthewrobertson) [reply src/cloud_events.ts:16]: Alternatively, as described below, pull the re-exported interface from `./functions`

```
import {CloudEvent} from './functions';
```

File: docs/generated/api.json [hunk 8]
```diff
@@ -223,8 +223,8 @@
                 },
                 {
                   "kind": "Reference",
-                  "text": "CloudEventsContext",
-                  "canonicalReference": "@google-cloud/functions-framework!CloudEventsContext:interface"
+                  "text": "CloudEvent",
+                  "canonicalReference": "cloudevents!CloudEvent:class"
```
Comment (matthewrobertson) [inline docs/generated/api.json:227]: This should be an `:interface`

**Comment by matthewrobertson** [comment]
We should pull in an updated version of the CESDK that includes the changes from https://github.com/cloudevents/sdk-javascript/pull/446

**Review by matthewrobertson — CHANGES_REQUESTED** [review]

**Comment by grant** [comment]
> We should pull in an updated version of the CESDK that includes the changes from [cloudevents/sdk-javascript#446](https://github.com/cloudevents/sdk-javascript/pull/446)

Alright by me. I've asked lance to see if we can get this in.

File: docs/generated/api.json (hunk 8 above)
Comment (grant) [reply docs/generated/api.json:227]: Looks like CI didn't catch this, thanks!

Fixing CI with https://github.com/GoogleCloudPlatform/functions-framework-nodejs/pull/401.

**Comment by grant** [comment]
Thanks for the review @matthewrobertson. PTAL :)

**Review by matthewrobertson — APPROVED** [review]

**Comment by matthewrobertson** [comment]
LGTM

**Comment by grant** [comment]
Merging with Matt's LGTM. Annie already reviewed the other PR.

**Pull request merged**